packages/server-wallet/src/wallet/index.ts

```diff
@@ -300,6 +335,15 @@ export class Wallet implements WalletInterface, ChainEventSubscriberInterface {
       });
     };
 
+    // FIXME: This is just to get existing joinChannel API pattern to keep working
+    /* eslint-disable-next-line */
+    const {objectiveId} = _.find(
+      this.store.objectives,
+      o => o.type === 'OpenChannel' && o.data.targetChannelId === channelId
+    )!;
```

Comment:
We expect an objective to exist because `pushMessage` would have stored it earlier. In the test cases which do not use the public API we manually insert the objective into the store, too.